File: Data Structures & Algorithms/binary-tree-right-side-view/submission-0.py
from collections import deque

# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        res = []
        q = deque()
        if root:
            q.append(root)
        while len(q) > 0:
            right = True
            for i in range(len(q)):
                curr = q.popleft()
                res.append(curr.val) if right == True else None
                right = False
                if curr.right:
                    q.append(curr.right)
                if curr.left:
                    q.append(curr.left)
        return res

File: Data Structures & Algorithms/find-pivot-index/submission-0.ts
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    pivotIndex(nums: number[]): number {
        let left: number = 0;
        let right: number = nums.reduce((acc, curr) => acc + curr, 0)

        for (let i = 0; i < nums.length; i++) {
            right -= nums[i]
            if (right === left) return i
            left += nums[i]
        }
        return -1
    }
}
